src/pybind/mgr/dashboard/frontend/src/app/ceph/smb/smb-cluster-form/smb-cluster-form.component.ts

```diff
@@ -59,31 +64,74 @@ export class SmbClusterFormComponent extends CdForm implements OnInit {
     private orchService: OrchestratorService,
     private modalService: ModalCdsService,
     private taskWrapperService: TaskWrapperService,
-    private router: Router
+    private router: Router,
+    private route: ActivatedRoute
   ) {
     super();
+
     this.resource = $localize`Cluster`;
     this.modalData$ = this.smbService.modalData$;
   }
+
   ngOnInit() {
-    this.action = this.actionLabels.CREATE;
+    if (this.router.url.startsWith(`/cephfs/smb/${URLVerbs.EDIT}`)) {
+      this.isEdit = true;
+    }
     this.smbService.modalData$.subscribe((data: DomainSettings) => {
       this.domainSettingsObject = data;
       this.smbForm.get('domain_settings').setValue(data?.realm);
     });
+
     this.createForm();
+    if (this.isEdit) {
+      this.action = this.actionLabels.EDIT;
+      this.route.params.subscribe((params: { cluster_id: string }) => {
+        this.cluster_id = params.cluster_id;
+      });
+
+      this.smbService.getCluster(this.cluster_id).subscribe((resp: SMBCluster) => {
+        this.clusterResponse = resp;
+        this.smbForm.get('cluster_id').setValue(this.clusterResponse.cluster_id);
+        this.smbForm.get('auth_mode').setValue(this.clusterResponse.auth_mode);
+
+        const responseClustering = this.clusterResponse.clustering;
+        const clusteringValue =
+          typeof responseClustering === 'object'
+            ? Object.values(responseClustering).find((value) => value) || ''
+            : responseClustering || '';
+
+        const upperFirstPipe = new UpperFirstPipe();
```

Comment:
push all consts in beginning of functions /if-else clause for better arrangement.
same comment for below or above